File: doc-examples/mongo-example-java/src/main/java/example/ManufacturerRepository.java

package example;

import io.micronaut.data.mongo.annotation.MongoRepository;
import io.micronaut.data.repository.GenericRepository;
import org.bson.types.ObjectId;

@MongoRepository
public interface ManufacturerRepository extends GenericRepository<Manufacturer, ObjectId> {
    Manufacturer findByName(String name);

    Manufacturer save(String name);
}

File: doc-examples/mongo-example-python/src/main/python/example/ManufacturerRepository.py
from micronaut.data.mongodb.annotation import MongoRepository
from micronaut.data.repository import GenericRepository
from org.bson.types import ObjectId

from example.Manufacturer import Manufacturer


@MongoRepository
class ManufacturerRepository(GenericRepository[Manufacturer, ObjectId]):
    def findByName(self, name: str) -> Manufacturer: ...

    def save(self, name: str) -> Manufacturer: ...
